tests/test_api.rs

```diff
@@ -259,6 +269,19 @@ fn array_buffer() {
     assert_eq!(84, bs.byte_length());
     assert_eq!(false, bs.is_shared());
 
+    let capacity = 100;
+    let mut data: Box<Vec<u8>> = Box::new(Vec::with_capacity(capacity));
+    let data_ptr = Box::into_raw(data);
+    let mut bs = v8::ArrayBuffer::new_backing_store_from_raw(
+      data_ptr as *mut std::ffi::c_void,
+      capacity,
+      deleter_callback,
+    );
+    assert_eq!(100, bs.byte_length());
+    assert_eq!(false, bs.is_shared());
```

Comment:
@piscisaureus I tried with just `Vec::with_capacity(capacity)` but then in deleter when I tried to `Vec::from_raw_parts(...)` I got segfault:
```
test_api-d423bd81ca175a6c(15580,0x70000d7f8000) malloc: *** error for object 0x7fe13da20fd0: pointer being freed was not allocated
test_api-d423bd81ca175a6c(15580,0x70000d7f8000) malloc: *** set a breakpoint in malloc_error_break to debug
```
Using `Box` works though

Reply:
A boxed vec is far from ideal though. It's a heap allocation to hold a pointer to another heap allocation. Maybe Vec::into_boxed_slice could help here.
Let's discuss Monday as I'm not working today.